cmd/metal-api/internal/service/v1/network.go

```diff
@@ -15,15 +15,17 @@ type NetworkBase struct {
-	PrivateSuper               bool     `json:"privatesuper" description:"if set to true, this network will serve as a partition's super network for the internal machine networks,there can only be one privatesuper network per partition"`
-	Underlay                   bool     `json:"underlay" description:"if set to true, this network can be used for underlay communication"`
-	Vrf                        *uint    `json:"vrf" description:"the vrf this network is associated with" optional:"true"`
-	VrfShared                  *bool    `json:"vrfshared" description:"if set to true, given vrf can be used by multiple networks, which is sometimes useful for network partitioning (default: false)" optional:"true"`
-	ParentNetworkID            *string  `json:"parentnetworkid" description:"the id of the parent network" optional:"true"`
-	AdditionalAnnouncableCIDRs []string `json:"additionalAnnouncableCIDRs,omitempty" description:"list of cidrs which are added to the route maps per tenant private network, these are typically pod- and service cidrs, can only be set for private super networks"`
+	Prefixes                   []string                `json:"prefixes" modelDescription:"a network which contains prefixes from which IP addresses can be allocated" description:"the prefixes of this network"`
+	DestinationPrefixes        []string                `json:"destinationprefixes" modelDescription:"prefixes that are reachable within this network" description:"the destination prefixes of this network"`
+	DefaultChildPrefixLength   metal.ChildPrefixLength `json:"defaultchildprefixlength" description:"if privatesuper, this defines the bitlen of child prefixes per addressfamily if not nil" optional:"true"`
+	Nat                        bool                    `json:"nat" description:"if set to true, packets leaving this network get masqueraded behind interface ip"`
+	PrivateSuper               bool                    `json:"privatesuper" description:"if set to true, this network will serve as a partition's super network for the internal machine networks,there can only be one privatesuper network per partition"`
+	Underlay                   bool                    `json:"underlay" description:"if set to true, this network can be used for underlay communication"`
+	Vrf                        *uint                   `json:"vrf" description:"the vrf this network is associated with" optional:"true"`
+	VrfShared                  *bool                   `json:"vrfshared" description:"if set to true, given vrf can be used by multiple networks, which is sometimes useful for network partitioning (default: false)" optional:"true"`
+	ParentNetworkID            *string                 `json:"parentnetworkid" description:"the id of the parent network" optional:"true"`
+	AddressFamilies            metal.AddressFamilies   `json:"addressfamilies" description:"the addressfamilies in this network, either IPv4 or IPv6 or both"`
```

Comment:
Better introduce a AF Struct which contains defaultchildprefixlenght and usage, deprecate usage

Reply:
This was not implemented because of too much trouble